ibazel/profiler/profiler.js

```diff
@@ -0,0 +1,123 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+(function(){
+  function setOption(options, name, value) {
+    if (typeof value === 'undefined') {
+      return;
+    }
+    if (!isNaN(+value)) {
+      value = +value;
+    }
+    return options[name] = value;
+  };
+
+  function extractOptions() {
+    var element, keyAndValue, m, mm, options, pair, src, _i, _j, _len, _len1, _ref, _ref1;
+    _ref = window.document.getElementsByTagName('script');
+    for (_i = 0, _len = _ref.length; _i < _len; _i++) {
+      element = _ref[_i];
+      if ((src = element.src) && (m = src.match(/^[^:]+:\/\/(.*)\/z?profiler\.js(?:\?(.*))?$/))) {
+        options = {};
+        options.https = src.indexOf("https") === 0;
+        if (mm = m[1].match(/^([^\/:]+)(?::(\d+))?$/)) {
+          options.host = mm[1];
+          if (mm[2]) {
+            options.port = parseInt(mm[2], 10);
+          }
+        }
+        if (m[2]) {
+          _ref1 = m[2].split('&');
+          for (_j = 0, _len1 = _ref1.length; _j < _len1; _j++) {
+            pair = _ref1[_j];
+            if ((keyAndValue = pair.split('=')).length > 1) {
+              setOption(options, keyAndValue[0].replace(/-/g, '_'), keyAndValue.slice(1).join('='));
+            }
+          }
+        }
+        options.url = (options.https ? "https" : "http") + "://" + options.host + ":" + options.port + "/profiler-event";
+        return options;
+      }
+    }
+    return null;
+  };
+
+  const profilerOptions = extractOptions();
+  if (!profilerOptions) {
+    this.console.error("iBazel profiler disabled because it could not find its own <SCRIPT> tag");
+    return;
+  }
+
+  if (!navigator.sendBeacon) {
+    console.error("iBazel profiler disabled because beacon API is not available")
+    return;
+  }
+
+  function profilerEvent(eventType, eventData) {
+    if (!eventType || typeof eventType !== 'string') {
+      console.error("Invalid iBazel profiler event type", eventType);
+      return;
+    }
+    if (typeof eventData === 'object') {
+      eventData = JSON.stringify(eventData)
+    }
+    const event = { type: eventType, time: Date.now() };
+    event.elapsed = event.time - window.performance.timing.navigationStart;
```

Comment:
this isn't really the elapsed time, it is the time at which the event occurred in ms relative to page load. Elapsed implies that you were monitoring the event from start to finish.

Reply:
Its elapsed time since the navigationStart event so it would be the remote elapsed time I suppose. Renamed to timeSinceNavigationStart but its a bit verbose. Have a better idea? In iBazel, this can value gets passed into remoteElapsedTime in the REMOTE_EVENT